app/controllers/facility_reservations_controller.rb

```diff
@@ -141,7 +141,7 @@ def create
   # GET /facilities/:facility_id/instruments/:instrument_id/reservations/:id/edit_admin
   def edit_admin
     @instrument  = current_facility.instruments.find_by_url_name!(params[:instrument_id])
-    @reservation = @instrument.reservations.find(params[:reservation_id])
+    @reservation = Reservation.where(product_id: @instrument.schedule.products.map(&:id)).find(params[:reservation_id])
```

Comment:
It might also be worth extracting this into a private method since it's used in several places in this controller.